### PR TITLE
Further redis TS conversion

### PR DESCRIPTION
**Pull request by Rory-Powell**
## Description
- Add additional TS conversions for redis
- Add `ioredis` types and integrate with `RedisWrapper`

### CODE REVIEW

**Review by Rory-Powell — COMMENTED** [review]
@mike12345567 the only change I'm concerned about in this one is the rename of `delete` to `deleteKey` in `cache/generic`. This had to be done as `delete` is a restricted function name. I think I've covered the usages but I'm not 100% sure, would be grateful if you could take a quick look to make sure there aren't any missed ones

File: packages/backend-core/src/cache/generic.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import BaseCache from "./base"
+
+const GENERIC = new BaseCache()
+
+export const CacheKeys = {
```
Comment (mike12345567) [inline packages/backend-core/src/cache/generic.ts:5]: Could possibly convert this and the TTL into real TS enums.

**Review by mike12345567 — APPROVED** [review]
LGTM!

**Review by mike12345567 — CHANGES_REQUESTED** [review]
Actually - had a think about the `deleteKey` changes - this will be a pretty huge change as the redis clients are returned from the `init` file and used all over the place.

For example have a look at `packages/backend-core/src/cache/user.js` - line 61 has a cache delete using the `redis.getUserClient()` function. Not sure how to account for this - changing usage of this could be quite difficult.

**Comment by mike12345567** [comment]
Closing as superseded by complete typescript refactor.

**Pull request closed without merging**